src/v8/v8.cc

```diff
@@ -92,6 +94,16 @@ class V8 : public WasmVm {
   FOR_ALL_WASM_VM_EXPORTS(_GET_MODULE_FUNCTION)
 #undef _GET_MODULE_FUNCTION
 
+  void terminate() override {
+    auto *store_impl = reinterpret_cast<wasm::StoreImpl *>(store_.get());
+    auto *isolate = store_impl->isolate();
+    isolate->TerminateExecution();
+    while (isolate->IsExecutionTerminating()) {
+      std::this_thread::yield();
+    }
+    integration()->trace("[host->vm] Terminated");
+  }
```

Comment:
As mentioned before, please move implementation outside of the class declaration.